### PR TITLE
Uses functions from asterism

### LINKED ISSUE #19: Factor out common functions in routines.py
Additionally, use functions from asterism (file_helpers and bagit_helpers) where applicable.

### PR DESCRIPTION
**Pull request by bonniegee**
Fixes #19

### CODE REVIEW

**Comment by helrond** [comment]
@bonniegee should this PR be against `development`?

File: package_bag/routines.py [hunk 1]
```diff
@@ -156,16 +157,9 @@ def create_package(self, bag, bag_json):
         mkdir(package_root)
         with open("{}.json".format(join(package_root, bag.bag_identifier)), "w",) as f:
             json.dump(bag_json, f, indent=4, sort_keys=True, default=str)
-        bag_tar_filename = "{}.tar.gz".format(bag.bag_identifier)
```
Comment (helrond) [inline package_bag/routines.py:159]: You actually need to make two tarfiles, one of the bag, and then one that contains the tarfile of the bag as well as the JSON. So you need to call `make_tarfile` twice.

File: package_bag/routines.py [hunk 2]
```diff
@@ -26,7 +27,7 @@ def run(self):
             try:
                 bag_id = self.unpack_rename(bag, settings.TMP_DIR)
                 bag_path = join(settings.TMP_DIR, bag_id)
-                self.validate_structure(bag_path)
+                validate(bag_path)
                 self.validate_metadata(bag_path)
```
Comment (helrond) [inline package_bag/routines.py]: I'm wondering if `validate_metadata` can return the bag info metadata, so you don't have to instantiate the bag again in `get_data`?

**Review by helrond — CHANGES_REQUESTED** [review]

**Comment by helrond** [comment]
@bonniegee if you do `pre-commit install` you'll get the linting stuff happening in your git commit flow...

File: package_bag/routines.py (hunk 2 above)
Comment (bonniegee) [reply package_bag/routines.py]: I tried to implement this, but am running into issues where it's saving invalid bags to the database:
```bag_info = self.validate_metadata(bag_path)
                new_bag = Bag.objects.create(
                    original_bag_name=bag,
                    bag_identifier=bag_id,
                    bag_path=bag_path)
                self.get_data(bag, bag_info)```

Comment (helrond) [reply package_bag/routines.py]: @bonniegee take a look at what I did L77-80

Comment (bonniegee) [reply package_bag/routines.py]: Ah, okay--that makes sense

**Pull request merged**